include/nuttx/progmem.h

```diff
@@ -49,96 +49,119 @@ extern "C"
+ *   may be required.
+ *
+ * Returned Value:
+ *   Required granularity of a write operation for the FLASH memory.
  *
  ****************************************************************************/
 
-size_t up_progmem_pagesize(size_t page);
+size_t up_progmem_writegranularity(void);
 
 /****************************************************************************
  * Name: up_progmem_erasesize
  *
  * Description:
- *  Return erase block size. Must be a multiple of the read/write page size.
+ *   Return the erase size for a specified index. Will always be a multiple of
+ *   the write granularity.
+ *
+ * Input Parameters:
+ *   index - ordinal number of the set of erasable units of memory
+ *
+ * Returned Value:
+ *   Erase size for erasable unit of memory, SIZE_MAX if index is invalid.
  *
  ****************************************************************************/
 
-size_t up_progmem_erasesize(size_t block);
+size_t up_progmem_erasesize(size_t index);
 
 /****************************************************************************
- * Name: up_progmem_getpage
+ * Name: up_progmem_getindex
  *
  * Description:
- *   Address to read/write page conversion
+ *   Get the index of memory, given a FLASH memory address
  *
  * Input Parameters:
  *  addr - Address with or without flash offset
- *         (absolute or aligned to page0)
+ *         (absolute or aligned to sector0)
  *
  * Returned Value:
- *   Page or negative value on error.
+ *   Index or negative value on error.
  *   The following errors are reported (errno is not set!):
  *
  *     -EFAULT: On invalid address
  *
  ****************************************************************************/
 
-ssize_t up_progmem_getpage(size_t addr);
+ssize_t up_progmem_getindex(size_t addr);
 
 /****************************************************************************
  * Name: up_progmem_getaddress
  *
  * Description:
- *   Read/write page to address conversion
+ *   Get the base address of a unit of memory, given an index
  *
  * Input Parameters:
- *   page - page index
+ *   index - ordinal number of the set of erasable units of memory
  *
  * Returned Value:
- *   Base address of given page, SIZE_MAX if page index is not valid.
+ *   Base address of given section, SIZE_MAX if index is not valid.
  *
  ****************************************************************************/
 
-size_t up_progmem_getaddress(size_t page);
+size_t up_progmem_getaddress(size_t index);
 
 /****************************************************************************
- * Name: up_progmem_eraseblock
+ * Name: up_progmem_erase
  *
  * Description:
- *   Erase selected erase block.
+ *   Erase a specified unit of memory, given an index
  *
  * Input Parameters:
- *   block - The erase block index to be erased.
+ *   index - ordinal number of the set of erasable units of memory
  *
  * Returned Value:
- *   block size or negative value on error.
+ *   section size or negative value on error.
```

Comment:
```suggestion
 *   The unit of memory's size or negative value on error.
```
return type needs to be ssize_t if negative return or use 0